File: src/main/java/src/Calendario/Horario.java
package Calendario;

import java.util.ArrayList;

public class Horario {
	
	protected ArrayList<dias> dia;
	private String hora_inicio;
	private String hora_Fin;
	public enum dias {lunes, martes, miercoles, jueves, viernes, sabado, domingo};
	
	
	public Horario(ArrayList<dias> dia,String hora_inicio,String hora_Fin) {
		
		this.dia = dia;
		this.hora_inicio = hora_inicio;
		this.hora_Fin = hora_Fin;
		
	}
	
	
	public ArrayList<dias> getDia() {
        return dia;
    }

    public void setDia(ArrayList<dias> dia) {
        this.dia = dia;
    }

    public String getHora_inicio() {
        return hora_inicio;
    }

    public void setHora_inicio(String hora_inicio) {
        this.hora_inicio = hora_inicio;
    }

    public String getHora_Fin() {
        return hora_Fin;
    }

    public void setHoraFinal(String hora_Fin) {
        this.hora_Fin = hora_Fin;
    }
    
    public String toString() {
    	return dia + " de " + hora_inicio + " a " + hora_Fin;
    }
}

File: src_py/Calendario/Horario.py
from typing import List
from enum import Enum

class Horario:
    class Dias(Enum):
        LUNES = 'lunes'
        MARTES = 'martes'
        MIERCOLES = 'miercoles'
        JUEVES = 'jueves'
        VIERNES = 'viernes'
        SABADO = 'sabado'
        DOMINGO = 'domingo'

    def __init__(self, dia: List[Dias], hora_inicio: str, hora_fin: str):
        self.__dia = dia
        self.__hora_inicio = hora_inicio
        self.__hora_fin = hora_fin

    def getDia(self):
        return self.__dia

    def setDia(self, dia):
        self.__dia = dia

    def getHora_inicio(self) :
        return self.__hora_inicio

    def setHora_inicio(self, hora_inicio):
        self.__hora_inicio = hora_inicio

    def getHora_fin(self):
        return self.__hora_fin

    def setHora_fin(self, hora_fin):
        self.__hora_fin = hora_fin

    def __str__(self):
        dias_str = ', '.join(day.name for day in self.__dia)
        return f'Los días [{dias_str}] de {self.__hora_inicio} a {self.__hora_fin}'

